syntax = "proto3";

package api;
import "diff.proto";

message Kim {
	option (diffable) = true;

	int64 id = 1;

	optional string name = 2;

	message TaskEntry {
		int64 id = 1;
		int32 curr_pos = 2;
		int32 swap_pos = 3;

		int64 task_type = 4 [(in) = true];
		string tag = 5 [(in) = true];
	}
	repeated TaskEntry tasks = 3;

	message GroupEntry {
		int64 id = 1;

		optional int64 start_time = 2;
		optional int64 end_time = 3;
		optional int64 duration = 4;
		optional bool is_virtual = 5;
		optional bool is_exam = 6;
		optional bool deleted = 7;
	}
	repeated GroupEntry groups = 4;
}

// Route /kim/list
message KimListResponse {
	repeated Kim kims = 1;
}

// Route /kim/list
// message UserKimListResponse {
// 	repeated Kim kim = 1;
// }

// message TaskAnswer {
// 	int64 pos = 1;
// 	bytes answer = 2;
// 	int64 submit_time = 3;
// }

// Route /kim/user/answers
// message KimUserAnswerListResponse {
// 	repeated TaskAnswer	answers = 1;
// }
